connectivity/tests/perfpod.go

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/defaults"
+)
+
+const (
+	messageSize = 1024
+)
+
+// Network Performance
+func NetperfPodtoPod(n string) check.Scenario {
+	return &netPerfPodtoPod{
+		name: n,
+	}
+}
+
+type netPerfPodtoPod struct {
+	name string
+}
+
+func (s *netPerfPodtoPod) Name() string {
+	tn := "perf-pod-to-pod"
+	if s.name == "" {
+		return tn
+	}
+	return fmt.Sprintf("%s:%s", tn, s.name)
+}
+
+func (s *netPerfPodtoPod) Run(ctx context.Context, t *check.Test) {
+	for _, c := range t.Context().PerfClientPods() {
+		c := c
+		for _, server := range t.Context().PerfServerPod() {
+			action := t.NewAction(s, "iperf-tcp", &c, server)
+			action.CollectFlows = false
+			action.Run(func(a *check.Action) {
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_RR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_CRR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "TCP_STREAM", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "UDP_RR", a, t.Context().PerfResults, 1)
+				netperf(ctx, server.Pod.Status.PodIP, c.Pod.Name, "UDP_STREAM", a, t.Context().PerfResults, 1)
+			})
+		}
+	}
+}
+
+func netperf(ctx context.Context, sip string, podname string, test string, a *check.Action, result map[check.PerfTests]check.PerfResult, samples int) {
+	// Define test about to be executed and from which pod
+	k := check.PerfTests{
+		Pod:  podname,
+		Test: test,
+	}
+
+	// Allow the user to override aspects of the test
+	iteration, _ := strconv.Atoi(os.Getenv("samples"))
+	duration, _ := strconv.Atoi(os.Getenv("duration"))
+	if samples < iteration {
+		samples = iteration
+	}
+	if duration <= 0 {
+		duration = defaults.PerfTestDuration
+	}
+	metric := string("OP/s")
+	if strings.Contains(test, "STREAM") {
+		metric = "Mb/s"
+	}
+
+	exec := []string{"/usr/local/bin/netperf", "-H", sip, "-l", fmt.Sprintf("%d", duration), "-t", test, "--", "-R", "1", "-m", fmt.Sprintf("%d", messageSize)}
+	//  recv socketsize		send socketsize 	msg size|okmsg	duration	value
+	r := regexp.MustCompile(`\s+\d+\s+\d+\s+(\d+|\S+)\s+(\S+|\d+)\s+(\S+)+\s+(\S+)?`)
+	values := []float64{}
+	// Result data
+	for i := 0; i < samples; i++ {
+		a.ExecInPod(ctx, exec)
+		d := r.FindStringSubmatch(a.CmdOutput())
+		if len(d) < 5 {
+			a.Fatal("Unable to process netperf result")
+		}
+		nv := ""
+		if len(d[len(d)-1]) > 0 {
+			nv = d[len(d)-1]
+		} else {
+			nv = d[len(d)-2]
+		}
+		f, _ := strconv.ParseFloat(nv, 64)
```

Comment:
Generally we should avoid hiding ignoring because it can result in really weird behavior without signal to the user as to what went wrong. Is this something that's totally unexpected? I don't have enough context about netperf, etc to tell whether there's a need to be defensive or not.

Reply:
This could leave to unwanted behavior. If the regex doesn't pick up the right value -- ie something change in netperf (highly unlikely).